AntiCovid/app/src/main/java/com/swtug/anticovid/view/profile/ProfileFragment.kt

```diff
@@ -65,10 +108,81 @@ class ProfileFragment : BaseFragment() {
             }
         }
 
-        btnlogout.setOnClickListener {
-            PreferencesRepo.deleteUser(requireContext())
-            PreferencesRepo.deleteVaccination(requireContext())
-            findNavController().navigate(R.id.action_profileFragment_to_loginFragment)
+        btnedit.setOnClickListener {
+            storeNewUserData()
+
         }
     }
+
+    private fun checkEditTextInputs(): String? {
+
+        if (editTextName.text.isEmpty() ||
+            editTextSurname.text.isEmpty() ||
+            editTextPhoneNumber.text.isEmpty() ||
+            editTextAddress.text.isEmpty() ||
+            editTextSocialSecurityID.text.isEmpty()
+        ) {
+            return requireActivity().getString(R.string.error_empty_fields)
+        }
+
+        if (editTextSocialSecurityID.text.length != 10) {
+            return requireActivity().getString(R.string.error_secID_lenth)
+        }
+
+        return null;
+    }
+
+    private fun storeNewUserData() {
+        val oldUser = PreferencesRepo.getUser(requireContext())
+        val validValues = checkEditTextInputs()
+        if (validValues != null) {
+
+            Toast.makeText(
+                requireActivity(),
+                validValues,
+                Toast.LENGTH_LONG
+            ).show()
+
+        } else if (oldUser != null) {
+
+            val user = User(
+                oldUser.id,
+                editTextName.text.toString(),
+                editTextSurname.text.toString(),
+                oldUser.email,
+                editTextAddress.text.toString(),
+                editTextSocialSecurityID.text.toString(),
+                editTextPhoneNumber.text.toString(),
+                oldUser.password
+            )
+            PreferencesRepo.saveUser(requireContext(), user)
+
+            FirebaseRepo.updateUser(user, object : FirebaseListener {
+                override fun onSuccess(user: User?) {
+                    btnedit.isEnabled = true
+                    Toast.makeText(
+                        requireContext(),
+                        requireContext().getString(R.string.changes_saved),
+                        Toast.LENGTH_LONG
+                    ).show()
+                }
+
```

Comment:
It would make sense to only save the User in our cache, if Firebase returns a SUCCESS. Also a revert to old user data would then be cool to actually always show the correct data which comes from Firebase